tests/test_auto_tag.py

```
@@ -164,6 +164,164 @@ def slow_tagger(url):
+
+        # Shrink the dedicated pool so saturation is reachable in-test.
+        small_pool = ThreadPoolExecutor(max_workers=2, thread_name_prefix="test-at")
+        monkeypatch.setattr("server.services.auto_tag._executor", small_pool)
+        monkeypatch.setattr(
+            "server.services.auto_tag._thread_budget",
+            threading.BoundedSemaphore(2),
+        )
+
+        release = threading.Event()
+
+        def maybe_stuck_tagger(url):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `maybe_stuck_tagger`

(ANN202)